### PR TITLE
feat(core): preview template & snapshot import

### PR DESCRIPTION
**Pull request by EYHN**

### CODE REVIEW

**Comment by EYHN** [comment]
* **#8193** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8193?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @EYHN and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/pages/workspace/share/share-page.tsx [hunk 1]
```diff
@@ -57,13 +57,24 @@ export const SharePage = ({
...
     const searchParams = new URLSearchParams(location.search);
     const queryStringMode = searchParams.get('mode') as DocMode | null;
     if (queryStringMode && DocModes.includes(queryStringMode)) {
       setMode(queryStringMode);
     }
+    if (searchParams.has('isTemplate')) {
+      setIsTemplate(true);
+    } else {
+      setIsTemplate(false);
+    }
```
Comment (pengx17) [inline packages/frontend/core/src/pages/workspace/share/share-page.tsx]: `setIsTemplate(searchParams.has('isTemplate'))`

File: packages/frontend/core/src/pages/workspace/share/share-page.tsx [hunk 2]
```diff
@@ -57,13 +57,24 @@ export const SharePage = ({
...
+    const templateName = searchParams.get('templateName');
+    if (templateName) {
+      setTemplateName(templateName);
+    }
   }, [location.search]);
```
Comment (pengx17) [inline packages/frontend/core/src/pages/workspace/share/share-page.tsx]: I think this effect can be refactored to using useMemo

File: packages/frontend/core/src/modules/import-template/store/downloader.ts [hunk 3]
```diff
@@ -7,9 +7,12 @@ export class TemplateDownloaderStore extends Store {
...
   }
 
-  async download(workspaceId: string, docId: string) {
+  async download(
+    /* not support workspaceid for now */ _workspaceId: string,
+    docId: string
+  ) {
     const response = await this.fetchService.fetch(
-      `/api/workspaces/${workspaceId}/docs/${docId}`,
+      `https://affine.pro/templates/snapshots/${docId}.zip `,
```
Comment (pengx17) [inline packages/frontend/core/src/modules/import-template/store/downloader.ts:15]: In the next affine.pro update I will add the snapshot url to the query params

**Review by pengx17 — APPROVED** [review]

**Pull request merged**